comp/api/api/def/go.mod

```diff
@@ -0,0 +1,12 @@
+module github.com/DataDog/datadog-agent/comp/api/api/def
+
+go 1.21.0
+
+require go.uber.org/fx v1.18.2
```

Comment:
Not a blocker. But why is this require outside of the rest of requires?

Reply:
https://github.com/golang/go/issues/45965

go.mod splits the [indirect](https://github.com/DataDog/datadog-agent/blob/main/go.mod#L324) vs [direct](https://github.com/DataDog/datadog-agent/blob/main/go.mod#L124) required dependencies since 1.17+, this the only direct dependency